gammapy/datasets/map.py

```diff
@@ -586,9 +586,14 @@ def meta(self, value):
     # TODO: keep or remove?
     @property
     def background_model(self):
-        try:
-            return self.models[f"{self.name}-bkg"]
-        except (ValueError, TypeError):
+        if self.models is None:
+            return None
+        elif isinstance(self.models, DatasetModels):
+            if self.name in self.models.background_models.keys():
+                return self.models[self.models.background_models[self.name]]
+            else:
+                return None
+        else:
             pass
```

Comment:
```suggestion
        if self.models and self.name in self.models.background_models.keys():
                return self.models[self.models.background_models[self.name]]
```
this should be enough